mail_drop_target/models/mail_thread.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# © 2018 Therp BV <https://therp.nl>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from StringIO import StringIO
+from base64 import b64decode
+try:
+    from ExtractMsg import Message
+except ImportError:
+    Message = None
+from odoo import _, api, exceptions, models
+
+
+class MailThread(models.AbstractModel):
+    _inherit = 'mail.thread'
+
+    @api.model
+    def message_process_msg(
+            self, model, message, custom_values=None, save_original=False,
+            strip_attachments=False, thread_id=None,
+    ):
+        """Convert message to RFC2822 and pass to message_process"""
+        if not Message:
+            raise exceptions.UserError(
+                _('Install the msg-extractor library to handle .msg files')
+            )
+        message_msg = Message(StringIO(b64decode(message)))
```

Comment:
Unfortunately in newer versions of ExtractMsg, like 0.17, the inner workings of the constructor changed, meaning a StringIO object is no longer valid.:
```
  File "/home/odorboprd11/odoo/parts/runbot/runbot/static/build/05694-10-0-import-development-dc9dd7/parts/odoo/odoo/addons/mail_drop_target/models/mail_thread.py", line 26, in message_process_msg
    message_msg = Message(StringIO(b64decode(message)))
  File "/home/odorboprd11/odoo/parts/runbot/runbot/static/build/05694-10-0-import-development-dc9dd7/eggs/ExtractMsg-0.17-py2.7.egg/ExtractMsg.py", line 657, in __init__
    elif len(path) < 1536:
AttributeError: StringIO instance has no attribute '__len__'
```

Reply:
I submitted an issue also on the ExtractMsg project: https://github.com/mattgwwalker/msg-extractor/issues/44

Reply:
Retested and works fine now!